omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -82,6 +82,9 @@ public void setSessionLocation(Location sessionLocation) {
         }
         this.sessionLocation = sessionLocation;
         this.sessionLocationId = sessionLocation.getId();
+        if (userContext != null && userContext.getAuthenticatedUser() != null) {
+            userContext.setLocationId(sessionLocation.getLocationId());
```

Comment:
Why do you call getLocationId? Don't you already have the id?

Reply:
I forgot that one and just replaced by mistake. I have fixed it by now.